packages/cdktf/test/remote-state.test.ts

```diff
@@ -0,0 +1,341 @@
+        key: "version-1.tfstate",
+        region: "cn-beijing",
+        tablestoreEndpoint: "https://terraform-remote.cn-hangzhou.ots.aliyuncs.com",
+        tablestoreTable: "statelock",
+        accessKey: "ALICLOUD_ACCESS_KEY",
+        secretKey: "ALICLOUD_SECRET_KEY",
+        securityToken: "ALICLOUD_SECURITY_TOKEN",
+        ecsRoleName: "RAM Role Name",
+        acl: "object acl",
+        assumeRole: {
+          roleArn: "ALICLOUD_ASSUME_ROLE_ARN",
+          policy: "restrictive policy",
+          sessionName: "ALICLOUD_ASSUME_ROLE_SESSION_NAME",
+          sessionExpiration: 3600
+        },
+        encrypt: true,
+        endpoint: "ALICLOUD_OSS_ENDPOINT",
+        profile: "ALICLOUD_PROFILE",
+        sharedCredentialsFile: "ALICLOUD_SHARED_CREDENTIALS_FILE"
+    });
+
+    expect(Testing.synth(stack)).toMatchSnapshot();
+})
+
+test('pg', () => {
+    const app = Testing.app();
+    const stack = new TerraformStack(app, 'test');
+
+    new b.DataTerraformRemoteStatePg(stack, 'remote', {
+        connStr: "postgres://user:pass@db.example.com/terraform_backend",
+        schemaName: "terraform_remote_state",
+        skipSchemaCreation: true
+    });
+
+    expect(Testing.synth(stack)).toMatchSnapshot();
+})
+
+test('s3', () => {
+    const app = Testing.app();
+    const stack = new TerraformStack(app, 'test');
+
+    new b.DataTerraformRemoteStateS3(stack, 'remote', {
+        bucket: "mybucket",
+        key: "path/to/my/key",
+        region: "us-east-1",
+        endpoint: "AWS_S3_ENDPOINT",
+        encrypt: true,
+        acl: "Canned ACL",
+        accessKey: "AWS_ACCESS_KEY_ID",
+        secretKey: "AWS_SECRET_ACCESS_KEY",
+        kmsKeyId: "ARN of a KMS Key",
+        dynamodbTable: "DynamoDB table",
+        profile: "AWS_PROFILE",
+        sharedCredentialsFile: "~/.aws/credentials",
+        token: "AWS_SESSION_TOKEN",
+        roleArn: "role to be assumed",
+        assumeRolePolicy: "permissions for assuming role",
+        externalId: "external ID",
+        sessionName: "role session name",
+        workspaceKeyPrefix: "env:",
+        dynamodbEndpoint: "AWS_DYNAMODB_ENDPOINT",
+        iamEndpoint: "AWS_IAM_ENDPOINT",
+        stsEndpoint: "AWS_STS_ENDPOINT",
+        forcePathStyle: false,
+        skipCredentialsValidation: false,
+        skipMetadataApiCheck: true,
+        sseCustomerKey: "AWS_SSE_CUSTOMER_KEY",
+        maxRetries: 5
+    });
+
+    expect(Testing.synth(stack)).toMatchSnapshot();
+})
+
+test('s3 with options', () => {
+    const app = Testing.app();
+    const stack = new TerraformStack(app, 'test');
+
+    new b.DataTerraformRemoteStateS3(stack, 'remote', {
+        bucket: "mybucket",
+        key: "path/to/my/key"
+    }, {
+        workspace: "my_workspace",
+        defaults: {
+            someProp: 'some_value'
+        }
+    });
+
+    expect(Testing.synth(stack)).toMatchSnapshot();
+})
+
+// This test is currently too sensitive to the environment it is ran in
```

Comment:
Let me quickly check what the issue is with https://github.com/hashicorp/terraform-cdk/issues/256

Reply:
Well, let's go with `getString` for now. For some reason resolving the `Intrinsic` value doesn't work as expected. However, we don't need this here anyway.